build/configs/artik053/hello-protected/defconfig

```diff
@@ -0,0 +1,863 @@
+# CONFIG_ARMV7R_TOOLCHAIN_BUILDROOT is not set
+# CONFIG_ARMV7R_TOOLCHAIN_CODESOURCERYL is not set
+CONFIG_ARMV7R_TOOLCHAIN_GNU_EABIL=y
+# CONFIG_ARMV7R_TOOLCHAIN_GNU_OABI is not set
+# CONFIG_ARMV7R_HAVE_DECODEFIQ is not set
+
+#
+# S5J Configuration Options
+#
+CONFIG_ARCH_CHIP_S5JT200=y
+CONFIG_S5J_S5JT200=y
+
+#
+# S5J Peripheral Support
+#
+CONFIG_S5J_HAVE_ADC=y
+CONFIG_S5J_HAVE_DMA=y
+CONFIG_S5J_HAVE_I2C=y
+CONFIG_S5J_HAVE_I2S=y
+CONFIG_S5J_HAVE_MCT=y
+CONFIG_S5J_HAVE_PWM0=y
+CONFIG_S5J_HAVE_PWM1=y
+CONFIG_S5J_HAVE_PWM2=y
+CONFIG_S5J_HAVE_PWM3=y
+CONFIG_S5J_HAVE_PWM4=y
+CONFIG_S5J_HAVE_PWM5=y
+CONFIG_S5J_HAVE_RTC=y
+CONFIG_S5J_HAVE_SFLASH=y
+CONFIG_S5J_HAVE_SPI=y
+CONFIG_S5J_HAVE_SSS=y
+CONFIG_S5J_HAVE_UART0=y
+CONFIG_S5J_HAVE_UART1=y
+CONFIG_S5J_HAVE_UART2=y
+CONFIG_S5J_HAVE_UART3=y
+CONFIG_S5J_HAVE_UART4=y
+CONFIG_S5J_HAVE_WATCHDOG=y
+# CONFIG_S5J_ADC is not set
+# CONFIG_S5J_DMA is not set
+CONFIG_S5J_I2C=y
+# CONFIG_S5J_I2S is not set
+# CONFIG_S5J_MCT is not set
+# CONFIG_S5J_TIMER0 is not set
+# CONFIG_S5J_TIMER1 is not set
+# CONFIG_S5J_TIMER2 is not set
+# CONFIG_S5J_TIMER3 is not set
+# CONFIG_S5J_UART_FLOWCONTROL is not set
+CONFIG_S5J_UART0=y
+CONFIG_S5J_UART1=y
+CONFIG_S5J_UART2=y
+CONFIG_S5J_UART3=y
+# CONFIG_S5J_UART3_FLOWCONTROL is not set
+CONFIG_S5J_UART4=y
+# CONFIG_S5J_PWM is not set
+# CONFIG_S5J_PWM0 is not set
+# CONFIG_S5J_PWM1 is not set
+# CONFIG_S5J_PWM2 is not set
+# CONFIG_S5J_PWM3 is not set
+# CONFIG_S5J_PWM4 is not set
+# CONFIG_S5J_PWM5 is not set
+# CONFIG_S5J_SSS is not set
+CONFIG_S5J_SPI=y
+# CONFIG_S5J_WATCHDOG is not set
+CONFIG_S5J_SFLASH=y
+# CONFIG_S5J_SENSOR_PPD42NS is not set
+
+#
+# Architecture Options
+#
+# CONFIG_ARCH_NOINTC is not set
+# CONFIG_ARCH_VECNOTIRQ is not set
+# CONFIG_ARCH_DMA is not set
+# CONFIG_ARCH_HAVE_IRQPRIO is not set
+# CONFIG_ARCH_L2CACHE is not set
+# CONFIG_ARCH_HAVE_COHERENT_DCACHE is not set
+# CONFIG_ARCH_HAVE_ADDRENV is not set
+# CONFIG_ARCH_NEED_ADDRENV_MAPPING is not set
+CONFIG_ARCH_HAVE_VFORK=y
+# CONFIG_ARCH_HAVE_MMU is not set
+CONFIG_ARCH_HAVE_MPU=y
+# CONFIG_ARCH_NAND_HWECC is not set
+# CONFIG_ARCH_HAVE_EXTCLK is not set
+# CONFIG_ARCH_HAVE_POWEROFF is not set
+CONFIG_ARCH_HAVE_RESET=y
+CONFIG_ARCH_USE_MPU=y
+# CONFIG_ARCH_STACKDUMP is not set
+# CONFIG_ENDIAN_BIG is not set
+# CONFIG_ARCH_IDLE_CUSTOM is not set
+# CONFIG_ARCH_HAVE_RAMFUNCS is not set
+# CONFIG_ARCH_HAVE_RAMVECTORS is not set
+
+#
+# Board Settings
+#
+CONFIG_BOARD_LOOPSPERMSEC=29100
+# CONFIG_ARCH_CALIBRATION is not set
+
+#
+# Interrupt options
+#
+CONFIG_ARCH_HAVE_INTERRUPTSTACK=y
+CONFIG_ARCH_INTERRUPTSTACK=0
+# CONFIG_ARCH_HAVE_HIPRI_INTERRUPT is not set
+
+#
+# Boot options
+#
+# CONFIG_BOOT_RUNFROMEXTSRAM is not set
+CONFIG_BOOT_RUNFROMFLASH=y
+# CONFIG_BOOT_RUNFROMISRAM is not set
+# CONFIG_BOOT_RUNFROMSDRAM is not set
+# CONFIG_BOOT_COPYTORAM is not set
+
+#
+# Boot Memory Configuration
+#
+CONFIG_RAM_REGIONx_START="0x02023800"
+CONFIG_RAM_REGIONx_SIZE="968704"
```

Comment:
Shouldn't it be 245760 ?

Reply:
Yes.. I will change it.

Reply:
Need to check, whether `CONFIG_RAM_REGIONx`  is only for kernel data + heap or it includes user ram as well

Reply:
According to up_allocate_heap(), below is the data:

`#define REGION_START (size_t)regionx_start[0]`
`#define REGION_SIZE  regionx_size[0]`
`#define REGION_END (REGION_START + REGION_SIZE)`

`void *regionx_start[CONFIG_MM_REGIONS] = {(void *)0x02023800};`
`size_t regionx_size[CONFIG_MM_REGIONS] = {1258290};`
`int regionx_heap_idx[CONFIG_MM_REGIONS] = {0};`

`DEBUGASSERT(ubase < (uintptr_t)REGION_END);`

The user bss also should lie withing `REGION_END`. So, I suppose that the `REGION_SIZE` is inclusive of user area also.

Reply:
Ok fine then.

But new change  `CONFIG_RAM_REGIONx_SIZE="1258290"` is confusing
As per artik code(s5j_boot.c), 
   
 ```
 TizenRT      0x02023800  0x0210FFFF  946(WBWA)                           
  WIFI         0x02110000  0x0215FFFF  320(NCNB)
```

I think only 946KB is available for TizenRT out 1.2MB ISRAM

Reply:
Ok. I reverted it to 968704.

Reply:
Yes. Kishore is correct. `CONFIG_RAM_REGIONx` includes user also.